src/dev/flang/ast/AstErrors.java

```diff
@@ -1038,6 +1038,39 @@ static String solutionDeclareReturnTypeIfResult(String name, int argCount)
     return solution;
   }
 
+  /**
+   * Suggest to a user that they might have called a feature with a wrong number of arguments,
+   * in the case that the called feature could not be found but there is a feature with the same
+   * name but a differing number of arguments.
+   */
+  static String solutionWrongArgumentNumber(FeatureName name,
+                                            List<FeatureAndOuter> l)
+  {
+    var solution = "";
+    var found = new List<FeatureName>();
+
+    for (var fo : l)
+      {
+        var f = fo._feature;
+        var fn = f.featureName();
+        if (f.isChoice() && !f.isBaseChoice())
+          {
+            /* ignore call to choice type */
+          }
+        else if (fn.equalsBaseName(name))
+          {
+            found.add(fn);
+          }
+      }
+
```

Comment:
For code maintenance, it would be much better not to have this logic duplicated far away from the call lookup code. Instead, you couldadd it to FeatureAndOuter, maybe add a variant of `filter` that does this. Use this variant in the existing FeatureAndOuter.filter and call the new variant in Call to pass this list to AstErrors.